plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -873,17 +873,6 @@ private BeginTableExecuteResult<ConnectorTableExecuteHandle, ConnectorTableHandl
         IcebergOptimizeHandle optimizeHandle = (IcebergOptimizeHandle) executeHandle.getProcedureHandle();
         Table icebergTable = catalog.loadTable(session, table.getSchemaTableName());
 
-        int tableFormatVersion = ((BaseTable) icebergTable).operations().current().formatVersion();
-        if (tableFormatVersion > OPTIMIZE_MAX_SUPPORTED_TABLE_VERSION) {
-            // Currently, Optimize would fail when position deletes files are present in Iceberg table
```

Comment:
Keep the condition, just remove this comment.
We don't know what Iceberg v3 brings us.